src/Confluent.Kafka.Avro/AvroDeserializer.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Copyright 2016-2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using Avro.Generic;
```

Comment:
The .NET version of the apache avro library will be **much** less battle tested than the java version. I hope it's not going to cause us too many problems.